controllers/remote/flags.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remote
+
+import (
+	"time"
+
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var (
+	restConfigTimeout time.Duration
+	restConfigQPS     float32
+	restConfigBurst   int
+)
+
+func init() {
+	// initialize for tests
+	restConfigQPS = 20
+	restConfigBurst = 30
+	restConfigTimeout = defaultClientTimeout
+}
+
+// AddRestConfigFlags adds the kube-api rest configuration flags to the flag set.
+func AddRestConfigFlags(fs *pflag.FlagSet) {
+	fs.Float32Var(&restConfigQPS, "kube-api-qps", 20,
+		"Maximum queries per second from the controller client to the Kubernetes API server.")
+
+	fs.IntVar(&restConfigBurst, "kube-api-burst", 30,
+		"Maximum number of queries that should be allowed in one burst from the controller client to the Kubernetes API server.")
+
+	fs.DurationVar(&restConfigTimeout, "kube-api-timeout", defaultClientTimeout,
+		"The maximum length of time to wait before giving up on a server request. A value of zero means no timeout.")
+}
+
+// DefaultRestConfig returns the runtime rest configuration with values set from the flags.
+func DefaultRestConfig() *rest.Config {
+	restConfig := ctrl.GetConfigOrDie()
+	restConfig.QPS = restConfigQPS
+	restConfig.Burst = restConfigBurst
```

Comment:
Is there a reason `Timeout` is not also defined here?

Reply:
The kubernetes clients used for the management clusters never had a timeout assigned in any of the various `main.go` and I didn't want to introduce a unintended change in behavior.

If we prefer consistency, I can add the timeout here as well.